### PR TITLE
Homework4: Krivosheev Konstantin

### PR DESCRIPTION
**Pull request by TexnoBY**
### PR checklist
Please make sure you've passed PR checklist
- [x] my PR passes CI checks (flake8 and pycodestyle)
- [x] my PR named according to guidelines.
- [x] my PR contains complete tasks descriptions for all tasks
- [x] my PR made from the fork feature branch to maxim homework# branch.
- [x] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore

### CODE REVIEW

File: src/homework4/task2.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+
+def cities_with_similar():
+    countries = {}
+    for _ in range(int(input())):
+        country, *cities = input().split()
+        countries[country] = cities
+    for _ in range(int(input())):
+        city = input()
+        for key, value in countries.items():
```
Comment (belovmd) [inline src/homework4/task2.py]: bad naming

File: src/homework4/task2.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            if city in value:
+                print(key)
```
Comment (belovmd) [inline src/homework4/task2.py]: function should return, not print

File: src/homework4/task2.py (hunk 1 above)
Comment (belovmd) [inline src/homework4/task2.py]: O(n**2) solution. has O(n) solution

File: src/homework4/task4.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+"""
+Даны два списка чисел.
+Посчитайте, сколько различных чисел входит только в один из этих списков
```
Comment (belovmd) [inline src/homework4/task4.py:3]: we have special sets operation for this task. it has one task solution

File: src/homework4/task5.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+
+def main():
+    languages = {}
+    students = int(input())
+    for _ in range(students):
+        for __ in range(int(input())):
+            lang = input()
+            languages[lang] = languages.get(lang, 0) + 1
+    print([key for key, value in languages.items() if value == students])
```
Comment (belovmd) [inline src/homework4/task5.py]: bad naming

Comment (belovmd) [inline src/homework4/task5.py]: can be solved with sets easier

**Review by belovmd — CHANGES_REQUESTED** [review]
please address next issues

File: src/homework4/task5.py (hunk 4 above)
Comment (TexnoBY) [reply src/homework4/task5.py]: solve

File: src/homework4/task2.py (hunk 1 above)
Comment (TexnoBY) [reply src/homework4/task2.py]: rename

File: src/homework4/task2.py (hunk 2 above)
Comment (TexnoBY) [reply src/homework4/task2.py]: add return list

File: src/homework4/task4.py (hunk 3 above)
Comment (TexnoBY) [reply src/homework4/task4.py:3]: use ^

File: src/homework4/task5.py (hunk 4 above)
Comment (TexnoBY) [reply src/homework4/task5.py]: rename

File: src/homework4/task2.py (hunk 1 above)
Comment (TexnoBY) [reply src/homework4/task2.py]: change my solution

File: .gitignore [hunk 5]
```diff
@@ -106,4 +106,5 @@ venv.bak/
 #additional
 .idea/
 .tox/
-.gitignore/
\ No newline at end of file
+.gitignore/
+/src/homework5/
```
Comment (belovmd) [inline .gitignore:110]: strange handling for redundant files )) branches would be easier

**Review by belovmd — COMMENTED** [review]
9

**Pull request closed without merging**